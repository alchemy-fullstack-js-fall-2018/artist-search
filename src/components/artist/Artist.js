import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';


export default function Arist ({ name, description, id }) {

  return (
    <div>
      <h3>{name} - {description} <Link to={ROUTES.ARTIST.linkTo(id)}>Get Details: </Link> </h3>
    </div>
  );
}
