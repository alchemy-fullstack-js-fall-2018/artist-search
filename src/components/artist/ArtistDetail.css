.artist {
  background-color: rgb(164, 212, 229);
  font-size: 20px;
  font-family: Arial, Helvetica, sans-serif;
}

li {
  margin: 30px;
  font-size: 25px;
}
