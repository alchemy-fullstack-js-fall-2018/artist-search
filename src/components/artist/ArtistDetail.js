import React, { Component } from 'react';
import { getArtist } from '../../services/musicSearch';
// import styles from './ArtistDetail.css';
import Songs from '../songs/Songs';

export default class ArtistDetail extends Component {
  state = {
    artist: {}
  };

  fetchArtist = () => {
    getArtist(this.props.match.params.id)
      .then(artist => this.setState({ artist }));
  };

  componentDidMount() {
    this.fetchArtist();
  }

  render() {
    const { id, name, songs } = this.state.artist;
    if(!songs) return null;

    return (
      // <div className={styles.artist}>
      <div>
        <h3>{name}</h3>
        <h4>BAND ID: {id}</h4>
        <Songs artist={name} songs={songs || 'Loading...'} />
      </div>
    );
  }
}
