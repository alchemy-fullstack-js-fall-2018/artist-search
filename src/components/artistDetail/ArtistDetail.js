import React, { Component, Fragment } from 'react';
import { getArtist } from '../../services/api';
import PropTypes from 'prop-types';
import Song from '../song/Song';

export default class Artist extends Component {
    static propTypes = {
        match: PropTypes.object.isRequired,
    };

    state = {
        artist: null
    };

    fetchArtist = () => {
        getArtist(this.props.match.params.id)
            .then(artist => this.setState({ artist }));
    };

    componentDidMount() {
        this.fetchArtist();
    }

    render() {
        if(!this.state.artist) return null;
        const { name, works } = this.state.artist;
        
        const songs = works.map(song => {
            return (
                <Song key={song.id}
                    artistName={name} 
                    songName={song.title} />
            );
        });

        return (
            <Fragment>
                <h2>Artist Detail for {name}</h2>
                {songs}
            </Fragment>
        );
    }
}
