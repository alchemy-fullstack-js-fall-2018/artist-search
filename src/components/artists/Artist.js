import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import PropTypes from 'prop-types';

const Artist = ({ name, description, id }) => {
  return (
    <div>
      <h3>
        {name} - {description}
      </h3>
      <Link to={ROUTES.ARTIST.linkTo(id)}>Get Works by {name}.</Link>
    </div>
  );
};

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  id: PropTypes.string.isRequired
};

export default Artist;
