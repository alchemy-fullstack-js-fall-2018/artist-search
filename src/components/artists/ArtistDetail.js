import React, { Component } from 'react';
import { getArtist } from '../../services/musicSearch';
import Works from '../works/Works';
import PropTypes from 'prop-types';

export default class ArtistDetail extends Component {
  static propTypes = {
    location: PropTypes.object.isRequired
  };

  state = {
    artist: {}
  };

  fetchArtist = () => {
    const id = this.props.location.pathname.slice(1).split('/')[1];
    getArtist(id).then(artist => this.setState({ artist }));
  };

  componentDidMount() {
    this.fetchArtist();
  }

  render() {
    const { name, description, works } = this.state.artist;
    if(works === undefined) return null;

    return (
      <div>
        <h3>{name}</h3>
        <h2>{description}</h2>
        <Works works={works} name={name} />
      </div>
    );
  }
}
