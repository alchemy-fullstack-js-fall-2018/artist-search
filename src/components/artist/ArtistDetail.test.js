import React from 'react';
import { shallow } from 'enzyme';
import ArtistDetail from './ArtistDetail';
import Artist from './Artist';
import { MemoryRouter } from 'react-router-dom';

describe('Arist Test', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Artist name="David Banner" id="abcd222" works="Hustlin"/>);
  });

  it('Renders a artist details', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders songs for the artist', () => {
    const wrapperWithROuter = shallow(
      <MemoryRouter>
        <ArtistDetail />
      </MemoryRouter>
    );
    expect(wrapperWithROuter.find('works')).toBeTruthy();
  });

});
