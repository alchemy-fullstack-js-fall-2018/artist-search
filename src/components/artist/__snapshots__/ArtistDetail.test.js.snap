// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arist Test Renders a artist details 1`] = `
<div>
  <h3>
    David Banner
     - 
     
    <Link
      replace={false}
      to="/artists/abcd222"
    >
      Get Details: 
    </Link>
     
  </h3>
</div>
`;
